### PR TITLE
implement `set_cookie_item` action

### PR DESCRIPTION
**Pull request by lxxxvi**
Hi 👋 

This PR addresses #1 (wohoo).

I may have taken a "overcomplicated" approach to build the Cookie String. I appreciate your feedback, letme know what/if I can improve anything, I'm eager to learn! Thank you.

### CODE REVIEW

File: src/actions/document.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+function CookieStringBuilder(streamElement: StreamElement) {
+  this.streamElement = streamElement
+  this.value = ""
+
+  // columns: [streamElementAttribute, cookieKey, isBooleanAttribute]
+  this.attributeToCookieKeyMapping = [
+    ["domain", "Domain", false],
+    ["path", "Path", false],
+    ["expires", "Expires", false],
+    ["max-age", "Max-Age", false],
```
[the remark this answers is not in the document]
Comment (lxxxvi) [reply src/actions/document.ts:12]: It's possible to set both, but `Max-Age` wins. `Max-Age` is a value in seconds and `Expires` is a datetime.

File: src/actions/document.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
+import { StreamElement, TurboStreamActions } from "@hotwired/turbo"
+
+function CookieStringBuilder(streamElement: StreamElement) {
```
[the remark this answers is not in the document]
Comment (lxxxvi) [reply src/actions/document.ts:3]: I like this idea, will do it!

**Comment by lxxxvi** [comment]
@marcoroth Thanks for your review, I appreciate it! I applied the suggested changes, you may have second look now 😉

File: src/actions/document.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (marcoroth) [reply src/actions/document.ts]: Ah interesting, TIL. But as I mentioned, I don't think it makes sense that we add any kind of validation for it. So this is fine as-is 👍🏼

**Review by marcoroth — APPROVED** [review]
This looks good to me!

Thanks for taking this on @lxxxvi! 🎉

**Pull request merged**